js_src/app.js

```diff
@@ -21,14 +21,14 @@ goog.require('goog.dom');
 goog.require('goog.events');
 goog.require('goog.ui.Component');
 goog.require('goog.ui.TableSorter');
-goog.require('app.MessageStructure');
+goog.require('app.pid');
 
 goog.provide('app.setup');
 
-var app = app || {}
```

Comment:
It doesn't look like app is being defined anywhere now. Although I understand why you've gone down this route.

See my suggestion here about pulling some stuff out into another file to avoid the circular dependencies:
https://groups.google.com/d/msg/open-lighting/4bgqRxhg6XQ/d2pCA7RFAgAJ

Reply:
Regardless where I put `var app = {};`, it complains about `app` already being defined in js_src/pid_display.js, where `goog.provide('app.pid');` is called. And it does indeed work without explicitly including it, so I wouldn't put too much effort into this.